### DISCUSSION TITLE
How can I detect the yellow lines array?

### DISCUSSION THREAD

**Discussion by lucasgamaleri** [haplokuon/netDxf#333] [Q&A] [answered]
I'm trying to detect the entities in a DXF file using the library. But those yellow lines are the only ones I can't get.

![image](https://user-images.githubusercontent.com/11754115/141299618-76181584-49b0-4424-9ad0-ddb47daa1751.png)

Using DWG TrueView LIST command I can get the following information about those lines. But I need some help to get this information into the netDxf library

```
Select objects: 1 found
Select objects:
                  ARRAY     Layer: "0"
                            Space: Model space
                   Handle = 2aa
      Array type:      Polar
      Base point:  X=2893.7490  Y=1635.4660  Z=   0.0000
    Center point:  X=2333.1716  Y=1495.9016  Z=   0.0000
                   Radius:   577.6896
                Direction:        CCW
          Number of items:         17
      Angle between items:          9
               Fill angle:        152
           Number of rows:          1
              Row spacing:   327.8672
  Row elevation increment:     0.0000
         Number of levels:          1
            Level spacing:     1.0000
           Rotate objects:        Yes
```

I can upload the original file into a Zip if needed.

Thank you very much in advance!

**Comment by haplokuon** [comment]
The lines you are looking for where created by an Array, which, at the moment, are not implemented in this library. Nonetheless, the internal block created as the graphical representation of the Array is loaded. You will have to look for an Insert that references a block named "*U#" where '#' is an integer.

**Reply by lucasgamaleri** [reply]
Let me understand. You mean I should search into the file for that insert and implement the array in my code as you did with the other entities?
If I'm right it will take a little more time than expected and probably it will be better to start a new discussion about this specific implementation.

**Comment by lucasgamaleri** [comment]
I found this fraction of code in the file. How should I look for every code number intepretation at the [Dxf format documentation](https://help.autodesk.com/view/OARX/2021/ENU/?guid=GUID-235B22E0-A567-4CF6-92D3-38A2306D73F3)?

I'm sure this code represents the array of lines as the numbers at the **line 3 and 5** are the same as the BasePoint.X and BasePoint.Y shown in my discussion opening.

```
*U2
 10
2893.749023267863
 20
1635.465995264704
 30
0.0
  0
INSERT
  5
2BB
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2863.131930915242
 20
1725.830538184337
 30
0.0
 41
1.0
 42
1.0
 43
1.0
 50
9.473684210526219
  0
INSERT
  5
2BC
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2818.058900878146
 20
1809.923218748571
 30
0.0
 41
1.0
 42
1.0
 43
1.0
 50
18.94736842105248
  0
INSERT
  5
2BD
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2759.759407919367
 20
1885.450207844871
 30
0.0
 41
1.0
 42
1.0
 43
1.0
 50
28.42105263157874
  0
INSERT
  5
2BE
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2689.823710231386
 20
1950.351326094856
 30
0.0
 41
1.000000000000001
 42
1.000000000000001
 43
1.000000000000001
 50
37.89473684210499
  0
INSERT
  5
2BF
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2610.159471338372
 20
2002.856240177254
 30
0.0
 41
1.000000000000001
 42
1.000000000000001
 43
1.000000000000001
 50
47.36842105263124
  0
INSERT
  5
2C0
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2522.939724008155
 20
2041.532752905995
 30
0.0
 41
1.000000000000001
 42
1.000000000000001
 43
1.000000000000001
 50
56.84210526315752
  0
INSERT
  5
2C1
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2430.543595582992
 20
2065.325869835981
 30
0.0
 41
1.000000000000001
 42
1.000000000000001
 43
1.000000000000001
 50
66.31578947368381
  0
INSERT
  5
2C2
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2335.491411587589
 20
2073.586576761389
 30
0.0
 41
1.000000000000001
 42
1.000000000000001
 43
1.000000000000001
 50
75.78947368421012
  0
INSERT
  5
2C3
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2240.375947818793
 20
2066.089543131351
 30
0.0
 41
1.000000000000002
 42
1.000000000000002
 43
1.000000000000002
 50
85.26315789473633
  0
INSERT
  5
2C4
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2147.791706180705
 20
2043.039268479986
 30
0.0
 41
1.000000000000002
 42
1.000000000000002
 43
1.000000000000002
 50
94.73684210526259
  0
INSERT
  5
2C5
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
2060.264143436107
 20
2005.0645042122
 30
0.0
 41
1.000000000000002
 42
1.000000000000002
 43
1.000000000000002
 50
104.2105263157889
  0
INSERT
  5
2C6
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
1980.180783329347
 20
1953.201102904398
 30
0.0
 41
1.000000000000002
 42
1.000000000000002
 43
1.000000000000002
 50
113.6842105263151
  0
INSERT
  5
2C7
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
1909.726091162396
 20
1888.86376294646
 30
0.0
 41
1.000000000000003
 42
1.000000000000003
 43
1.000000000000003
 50
123.1578947368414
  0
INSERT
  5
2C8
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
1850.821887275893
 20
1813.807439257508
 30
0.0
 41
1.000000000000003
 42
1.000000000000003
 43
1.000000000000003
 50
132.6315789473677
  0
INSERT
  5
2C9
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
1805.074924800064
 20
1730.079472690546
 30
0.0
 41
1.000000000000003
 42
1.000000000000003
 43
1.000000000000003
 50
142.1052631578939
  0
INSERT
  5
2CA
330
2A7
100
AcDbEntity
  8
0
  6
ByBlock
 62
     0
370
    -2
380
     1
100
AcDbBlockReference
  2
*U2
 10
1773.733061617818
 20
1639.963743911001
 30
0.0
 41
1.000000000000003
 42
1.000000000000003
 43
1.000000000000003
 50
151.5789473684202
  0
ENDBLK
```

**Comment by haplokuon** [comment, marked as answer]
An excerpt of a DXF has a very limited use specially with complex objects like the Array. Most of the time the necessary information is spread through the whole file.

In your case you are dealing with an Array. When an Array is build in AutoCad it creates a block named "U#" that holds the entities for each cell of the array, then it builds another block, also named "U#", with all references to that block (Inserts) ordered according to the properties of the Array, and then it inserts it into the drawing. So, you will need to look for an Insert in the document entities and then, inside the entities of the block associated with that insert, you will find a list of Insert entities that references another block in which list of entities you should find the lines.

When you load a DXF with this library you loose all information of the original array but the Insert entities and its blocks (its graphical representation) is loaded. The main problem of implementing the Array object is that its properties are stored in the DXF Objects section which is an undocumented hodgepodge, and the only way to try to make some sense out of it, is to manually track the handle references through it.

In general terms, I cannot tell you how to find a specific entity without having some kind of constraint to limit the search. You could search an entity knowing its handle (the best way), giving a unique color or layer, adding some kind of extended data to it, by its coordinates, etc. For an array object, as in your sample, you could use the handle or attach some extended data to it, since that information is stored in the Insert. In your first post the output of the list command shows the information of an array with handle "2aa", try to find the Insert with that handle. For this kind of operations, in the DxfDocument class, there is a method called "GetObjectByHandle", and then search through the tree of inserts.

Hopefully this will also clarify your other post.
